modules/imgcodecs/include/opencv2/imgcodecs.hpp

```diff
@@ -94,7 +94,8 @@ enum ImwriteFlags {
        IMWRITE_PAM_TUPLETYPE       = 128,//!< For PAM, sets the TUPLETYPE field to the corresponding string value that is defined for the format
        IMWRITE_TIFF_RESUNIT = 256,//!< For TIFF, use to specify which DPI resolution unit to set; see libtiff documentation for valid values
        IMWRITE_TIFF_XDPI = 257,//!< For TIFF, use to specify the X direction DPI
-       IMWRITE_TIFF_YDPI = 258 //!< For TIFF, use to specify the Y direction DPI
+       IMWRITE_TIFF_YDPI = 258, //!< For TIFF, use to specify the Y direction DPI
+       IMWRITE_JPEG2000_COMPRESSION = 512 //!< For JPEG2000, use to specify the target compression rate. A higher value means a smaller size. Default is 1.
```

Comment:
Initial increment was "16" between codecs: "(codec_index << 4) + prop_index"

Also we have plans to drop "Jasper" support completely (Jasper project is in abandoned state for several years) and replace in by OpenJPEG.
So, how is this property is mapped into OpenJPEG implementation?

Reply:
Thank you for your feedback!
Concerning the increment: I was just guessing that the increment is a power of 2 because the previous values were 64,128, and 256.
I can adjust this to 272, if you prefer.

Concerning an OpenJPEG-based implementation: I have not used this library so far, i.e. I cannot directly answer this question.
If you like, I can do some research and propose a OpenJPEG-based implementation.
If you anyway want to drop JasPer, there is not really a point in merging this PR, right?

What do you think?

Reply:
If it is Jasper specific behavior (without alternative in OpenJPEG) then it is better to avoid using such common name.

What is about this name?

`IMWRITE_JPEG2000_JASPER_COMPRESSION`

Reply:
I think this name is fine.
However, I am wondering if it makes sense to introduce this enum, if it will be deprecated soon.
I will implement the changes later this week, if we agreed on the details.